plan/outline.md

```diff
@@ -0,0 +1,73 @@
+# Docs Outline
+
+* What are web components
+  * Custom Elements
+  * Shadow DOM
+  * Standard JS Modules
+* Why web components
+  * What do we mean by interop?
+    * Interop with HTML, frameworks, the browser
+    * Interop with file formats: Markdown vs MDX, etc.
+    * Interoperable composition with slots
+    * Interop with the future
+  * Use cases
+    * Standalone components
+    * Component sets and design systems
+    * Applications
+* Related specs
+  * MDN links
+  * GitHub repos with proposals
+  * What's left, ie. WCCG reports
+* Using web components
+  * Basics
+    * Definining, the custom element registry, and the single tag name scope
+    * Upgrading, and `:defined`
+    * Children and slots
+    * Events
+  * Using web components in HTML and vanilla JS
+  * Styling
+    * CSS Custom Properties
+    * ::part()s
+  * Using web components with frameworks
+  * Using web components in...
+    * Markdown (how to import in various systems)
+    * WordPress
+* Writing web components
+  * Defining
+    * Custom element registries
+    * Self-registering vs not
+    * Defensive registration for special cases (CDN distribution)
+  * Lifecycle
+    * Basics: `observedAttributes`, `constructor`, `attributeChangedCallback`...
+    * Cleaning up resources in `disconnectedCallback`
+  * Shadow DOM
+    * attachShadow()
+    * open vs closed
+    * Encapsulation: DOM, events, CSS
+      * Just overview: more detail in Styling and Composition sections
+    * Slots
+  * Styling
+    * Encapsulation (more detail than)
+    * Shadow selectors: `:host`, `:host()`, and `::slotted()`
+    * CSS Custom Properties
+    * CSS Shadow Parts: `part` and `::part()`
+    * Constructible style sheets
+    * Theming
+      * Strategies for deep styling
+  * Dealing with data
+    * Making reactive properties with accessors
```

Comment:
```suggestion
    * Designing both a property & attribute API
    * Responding to attribute changes (`attributeChangedCallback`)
    * Responding to property changes (accessors)
    * Considerations about when to use methods
```

Reply:
I meant to call out accessors specifically here as that's a key technique to have reactive state outside of attributes.

Reply:
oh - I misread the diff... you just expanded it :)